### PR TITLE
[T4A3][T10-B2] Li Zhiyuan

### PR DESCRIPTION
**Pull request by lzyuan168**
inheritence

### CODE REVIEW

**Comment by lzyuan168** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -6,26 +6,25 @@
...
+   
 
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
+        super(address, isPrivate);
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:23]: Can push the validation to the constructor too if you need

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package seedu.addressbook.data.person;
+
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Contact.java:2]: Header comments?

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	@Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Address // instanceof handles nulls
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Contact.java:20]: should compare with an intenceof Contact.
Can override in subclasses to their respective type
If you override in subclasses make sure to override the hashCode() too

**Comment by nirandiw** [comment]
@lzyuan168 Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**